ols/app/models/config.py

```diff
@@ -121,6 +121,42 @@ def validate_yaml(self, disable_tls: bool = False) -> None:
             checks.file_check(self.tls_key_path, "OLS server certificate private key")
 
 
+class ProxyConfig(BaseModel):
+    """HTTPS Proxy configuration."""
+
+    proxy_url: Optional[AnyHttpUrl] = None
+    proxy_ca_cert_path: Optional[FilePath] = None
+
+    def __init__(self, data: Optional[dict] = None) -> None:
+        """Initialize configuration and perform basic validation."""
+        super().__init__()
+        if data is None:
+            return
+        self.proxy_url = data.get("proxy_url", None)
+        if self.proxy_url is None:
+            # check if HTTPS_PROXY is set in the environment
+            # and use it as a proxy URL
+            os_proxy = os.environ.get("HTTPS_PROXY")
+            if os_proxy:
+                self.proxy_url = os_proxy
+            else:
+                os_proxy = os.environ.get("https_proxy")
```

Comment:
Is there a reason not to follow the industry standard and check for a lowercase env variable? Is there env where user have no controll about this and it is set as lowercase?

Reply:
good point, removing uppercase env var.